R/module_nested_tabs.R

```diff
@@ -491,3 +458,124 @@ srv_teal_module.teal_module <- function(id,
     ...
   )
 }
+
+#' @keywords internal
+.modules_breadcrumb <- function(module) {
+  if (is.null(module$group)) {
+    breadcrumb_items <- c("Home", module$label)
+  } else {
+    breadcrumb_items <- c("Home", module$group, module$label)
+  }
+  paste(breadcrumb_items, collapse = " / ")
+}
+
+
+#' @keywords internal
+.teal_custom_nav_deps <- function() {
+  htmltools::htmlDependency(
+    name = "module-navigation",
+    version = utils::packageVersion("teal"),
+    package = "teal",
+    src = "module-navigation",
+    stylesheet = "module-navigation.css",
+    script = "module-navigation.js"
+  )
+}
+
+#' Create Bootstrap based Navigation for Teal Modules
+#'
+#' Generates a dropdown navigation interface that allows users to switch
+#' between different teal modules. The function creates both the navigation dropdown menu
+#' and the corresponding tab content containers.
+#'
+#' @details
+#' This function constructs a complete navigation system with the following components:
+#' - A dropdown button labeled "Modules" that reveals module options on hover
+#' - Individual navigation links for each module, grouped by their `group` attribute
+#' - Tab content containers that house the actual module UI elements
+#' - Automatic tab switching via Bootstrap's tab functionality
+#'
+#' The navigation leverages `shiny`'s built-in tab system by using the `shiny-tab-input` class,
+#' which automatically handles showing/hiding content when navigation links are clicked.
+#' Each module's UI is wrapped in a `.tab-pane` container with an ID that corresponds to
+#' the navigation link's `href` attribute.
+#'
+#' Module grouping is supported - when modules belong to different groups, visual separators
+#' and group labels are automatically inserted in the dropdown menu.
+#'
+#' @param ns (`function`) A namespace function from the shiny module it is used in.
+#' @param modules (`flat_teal_modules`) A teal modules object that has been flattened by `flat_teal_modules()`.
+#' @keywords internal
+.teal_custom_nav <- function(ns, modules) {
```

Comment:
Yup, with this change I wanted to highlight that `.teal_custom_nav` is unnecessary and the whole code can be moved to `ui_teal_module.teal_modules`.